src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] hashTable;
+    private int size;
+    private int capacity;
+    private float threshold;
+
+    public MyHashMap() {
+        hashTable = new Node[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        threshold = DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        putValue(getIndexOfBucket(key), key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getIndexOfBucket(key);
+        Node<K, V> currentNode = hashTable[hash];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
```

Comment:
```suggestion
    private static class Node<K, V> {
```
better move Node class to the bottom